### PR TITLE
Scp049 events 

### PR DESCRIPTION
**Pull request by Undid-Iridium**
https://github.com/Exiled-Team/Exiled-EA/pull/214

### CODE REVIEW

File: Exiled.Events/EventArgs/Scp049/ZombieStartConsumeEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    using System.Collections.Generic;
+
+    using API.Features;
+    using Interfaces;
+    using Mirror;
+
+    /// <summary>
+    ///     Contains all information before zombie consumes ragdoll.
+    /// </summary>
+    public class ZombieStartConsumeEventArgs : IPlayerEvent, IDeniableEvent
```
Comment (Thundermaker300) [inline Exiled.Events/EventArgs/Scp049/ZombieStartConsumeEventArgs.cs:19]: Rename to `ConsumingBodyEventArgs`?

File: Exiled.Events/EventArgs/Scp049/Sending049CallEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        /// <summary>
+        ///     Gets the player who is controlling SCP-049.
+        /// </summary>
+        public Player Player { get; }
+
+        /// <summary>
+        ///     Gets the player who's getting recalled.
+        /// </summary>
+        public NetworkReader Reader { get; }
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Scp049/Sending049CallEventArgs.cs:51]: Should be Player

File: Exiled.Events/EventArgs/Scp049/ConsumingBodyEventArgs.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            IsAllowed = isAllowed;
+        }
+
+        /// <summary>
+        /// Gets or sets error code to send back to client.
+        /// </summary>
+        public byte ErrorCode { get; set; }
+
+        /// <summary>
+        /// Gets or sets scp0492 ragdolls consumed thus far.
```
[the remark this answers is not in the document]
Comment (Undid-Iridium) [reply Exiled.Events/EventArgs/Scp049/ConsumingBodyEventArgs.cs:44]: image

[the remark this answers is not in the document]
Comment (Thundermaker300) [reply Exiled.Events/EventArgs/Scp049/ConsumingBodyEventArgs.cs]: Documentation man approves 👍

File: Exiled.Events/EventArgs/Scp049/DoctorSenseEventArgs.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        /// <param name="senseAbility"> Doctor's <see cref="Scp049SenseAbility"/> ability. </param>
+        /// <param name="isAllowed"><inheritdoc cref="DoctorSenseEventArgs.IsAllowed"/></param>
+        public DoctorSenseEventArgs(Player scp049, Player target, Scp049SenseAbility senseAbility, bool isAllowed = true)
+        {
+            Player = scp049;
+            Target = target;
+            SenseAbility = senseAbility;
+            IsAllowed = isAllowed;
+            Cooldown = 5f;
+            Duration = 20f;
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Scp049/DoctorSenseEventArgs.cs:34]: Could we not hsrdcoded them

**Review by louis1706 — CHANGES_REQUESTED** [review]
(I will make the changes when it got time)

File: Exiled.Events/EventArgs/Scp049/DoctorSenseEventArgs.cs (hunk 4 above)
Comment (Undid-Iridium) [reply Exiled.Events/EventArgs/Scp049/DoctorSenseEventArgs.cs:34]: Yeah that's fair when I'm home ill do that

**Comment by Undid-Iridium** [comment]
Since I can't seem to tag it, Needs testing.

**Comment by Undid-Iridium** [comment]
@xNexus-ACS said they'll do it, please refer to their PR's. Thanks

**Pull request closed without merging**